bundles/org.openhab.binding.twitter/src/main/java/org/openhab/binding/twitter/internal/TwitterHandler.java

```diff
@@ -0,0 +1,298 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.twitter.internal;
+
+import static org.openhab.binding.twitter.internal.TwitterBindingConstants.CHANNEL_LASTTWEET;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang.UnhandledException;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.twitter.internal.action.TwitterActions;
+import org.openhab.binding.twitter.internal.config.TwitterConfig;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import twitter4j.DirectMessage;
+import twitter4j.ResponseList;
+import twitter4j.Status;
+import twitter4j.StatusUpdate;
+import twitter4j.Twitter;
+import twitter4j.TwitterException;
+import twitter4j.TwitterFactory;
+import twitter4j.auth.AccessToken;
+
+/**
+ * The {@link TwitterHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Scott Hanson - Initial contribution
+ */
+
+@NonNullByDefault
+public class TwitterHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(TwitterHandler.class);
+
+    private @Nullable TwitterConfig config;
+    @SuppressWarnings("unused")
+    private @Nullable ScheduledFuture<?> refreshTask;
+
+    private static final int CHARACTER_LIMIT = 280;
+
+    private static @Nullable Twitter client = null;
+    boolean isProperlyConfigured = false;
+
+    public TwitterHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    // creates list of available Actions
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singletonList(TwitterActions.class);
+    }
+
+    @SuppressWarnings({ "null", "unused" })
+    @Override
+    public void initialize() {
+        config = getConfigAs(TwitterConfig.class);
+
+        try {
+            // create a New Twitter Client
+            client = createClient();
+            // verify client is valid
+            if (client != null) {
```

Comment:
The field can be null at this point, because it isn't inittialized in the constructor. So, you need to check if it is not null before accessing it: Example: if (job != null). But this would result in a compiler warning, as it is a field and could theoretically be modified by another thread exactly between the null check and the access in the next line. To mitigate this, you can store the field to a local variable before checking it:
```
Future<?> localPollingJob = pollingJob;
if (localPollingJob != null) {
    localPollingJob.cancel(true);
}
```